src/Catel.Core/Catel.Core.Shared/Services/ObjectIdGenerator.cs

```diff
@@ -51,12 +56,58 @@ public void ReleaseIdentifier(TUniqueIdentifier identifier)
             }
         }
 
+        /// <inheritdoc />
+        public TUniqueIdentifier GetUniqueIdentifierForInstance(TObjectType instance, bool reuse = false)
+        {
+            Argument.IsNotNull("instance", instance);
+
+            lock (_syncObj)
+            {
+                if (_allocatedUniqueIdentifierPerInstances != null && _allocatedUniqueIdentifierPerInstances.TryGetValue(instance, out var wrapper))
+                {
+                    return wrapper.UniqueIdentifier;
+                }
+
+                if (_allocatedUniqueIdentifierPerInstances == null)
```

Comment:
This can never be null, so we should remove the null-checks to make the code even faster.